projects/common-platform-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/PersonService.kt

```diff
@@ -27,113 +28,138 @@ class PersonService(
     private val staffRepository: StaffRepository,
     private val referenceDataRepository: ReferenceDataRepository,
     private val personAddressRepository: PersonAddressRepository,
-    private val osClient: OsClient
+    private val osClient: OsClient,
+    private val featureFlags: FeatureFlags
 ) : AuditableService(auditedInteractionService) {
 
     @Transactional
     fun insertPerson(defendant: Defendant, courtCode: String): InsertPersonResult =
-        audit(BusinessInteractionCode.INSERT_PERSON) { audit ->
+        if (featureFlags.enabled("common-platform-record-creation-toggle")) {
+            audit(BusinessInteractionCode.INSERT_PERSON) { audit ->
+                val result = insertPersonLogic(defendant, courtCode)
+                audit["offenderId"] = result.person.id!!
+                result
+            }
+        } else {
+            insertPersonLogic(defendant, courtCode)
+        }
 
-            val dateOfBirth = defendant.personDefendant?.personDetails?.dateOfBirth
-                ?: throw IllegalArgumentException("Date of birth not found in message")
+    fun insertPersonLogic(defendant: Defendant, courtCode: String): InsertPersonResult {
+        val dateOfBirth = defendant.personDefendant?.personDetails?.dateOfBirth
+            ?: throw IllegalArgumentException("Date of birth not found in message")
 
-            // Under 10 years old validation
-            dateOfBirth.let {
-                val age = Period.between(it, LocalDate.now()).years
-                require(age > 10) {
-                    "Date of birth would indicate person is under ten years old: $it"
-                }
+        // Under 10 years old validation
+        dateOfBirth.let {
+            val age = Period.between(it, LocalDate.now()).years
+            require(age > 10) {
+                "Date of birth would indicate person is under ten years old: $it"
             }
+        }
```

Comment:
Would it simplify things to move this out into the Handler, or into a "validate" method?

That way you could perform this validation regardless of the feature flag, and only call `insertPerson` if the flag is enabled - e.g. the Handler would become something like:

```kotlin
// this could potentially be moved to a validate message:
val dateOfBirth = requireNotNull(defendant.personDefendant?.personDetails?.dateOfBirth) { "Date of birth not found in message" }
val age = Period.between(dateOfBirth, LocalDate.now()).years
require(age > 10) { "Date of birth would indicate person is under ten years old: $it" }

// only need to check the flag in one place:
if (featureFlags.enabled("common-platform-record-creation-toggle")) {
    val (person, address) = personService.insertPerson(defendant, notification.message.hearing.courtCentre.code)
    notifier.caseCreated(person)
    address?.let { notifier.addressCreated(it) }
    telemetryService.trackEvent("PersonCreated", mapOf("hearingId" to ..., "defendantId" to ..., "crn" to ..., etc))
} else {
    telemetryService.trackEvent("SimulatedPersonCreated", mapOf("hearingId" to ..., "defendantId" to ...))
}
```

I'm possibly missing some important detail though, so please let me know if I am! 😄